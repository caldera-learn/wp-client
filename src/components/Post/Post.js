import React from 'react';
export const  Post = (props) =>  {
	return(
		<div>
			<h3>
				{props.post.title.rendered}
			</h3>
			<div>
				{props.post.content.rendered}
			</div>
		</div>
	);
};

