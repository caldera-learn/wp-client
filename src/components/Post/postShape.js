import PropTypes from 'prop-types';

export const postShape =
	PropTypes.shape({
		ID: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.number
		]),
		title: PropTypes.shape({
			rendered: PropTypes.string,
		}).isRequired,
		content: PropTypes.shape({
			rendered: PropTypes.string,
		}).isRequired,
	}).isRequired;
