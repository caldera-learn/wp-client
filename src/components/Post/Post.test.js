import React from 'react';
import {Post} from "./Post";
import renderer from 'react-test-renderer';
describe( 'Post component', () => {
	/** Mock WordPress post to test with**/
	const post = {
		title: {
			rendered: 'Hi Roy'
		},
		content: {
			rendered: 'Lorem ipsum, etc.'
		}
	};

	it( 'Matches snapshot with basic props', () => {
		const component = renderer.create(
			<Post
				post={post}
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});
});