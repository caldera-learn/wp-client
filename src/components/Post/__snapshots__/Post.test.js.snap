// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component Matches snapshot with basic props 1`] = `
<div>
  <h3>
    Hi Roy
  </h3>
  <div>
    Lorem ipsum, etc.
  </div>
</div>
`;
